teleport-server/templates/cloud-init.yaml.tpl

```diff
@@ -0,0 +1,175 @@
+
+      # Optional setting for configuring session recording. Possible values are:
+      #    "node"  : sessions will be recorded on the node level  (the default)
+      #    "proxy" : recording on the proxy level, see "recording proxy mode" in "Audit Log" section
+      #    "off"   : session recording is turned off
+      session_recording: "${teleport_session_recording}"
+
+    # This section configures the 'node service':
+    ssh_service:
+      # Turns 'ssh' role on. Default is 'yes'
+      enabled: yes
+
+      # IP and the port for SSH service to bind to.
+      listen_addr: 0.0.0.0:3022
+
+      # See explanation of labels in "Labeling Nodes" section below
+      labels:
+        role: teleport-server
+        aws_region: ${teleport_dynamodb_region}
+
+    # This section configures the 'proxy servie'
+    proxy_service:
+      # Turns 'proxy' role on. Default is 'yes'
+      enabled: yes
+
+      # SSH forwarding/proxy address. Command line (CLI) clients always begin their
+      # SSH sessions by connecting to this port
+      listen_addr: 0.0.0.0:3023
+
+      # Reverse tunnel listening address. An auth server (CA) can establish an
+      # outbound (from behind the firewall) connection to this address.
+      # This will allow users of the outside CA to connect to behind-the-firewall
+      # nodes.
+      tunnel_listen_addr: 0.0.0.0:3024
+
+      # The HTTPS listen address to serve the Web UI and also to authenticate the
+      # command line (CLI) users via password+HOTP
+      web_listen_addr: 0.0.0.0:443
+
+      # TLS certificate for the HTTPS connection. Configuring these properly is
+      # critical for Teleport security.
+      #https_key_file: /etc/letsencrypt/live/${teleport_domain_name}/privkey.pem
+      #https_cert_file: /etc/letsencrypt/live/${teleport_domain_name}/fullchain.pem
+  path: /etc/teleport.yaml
+- content: |
+    [Unit]
+    Description=Teleport SSH Service
+    After=network.target
+
+    [Service]
+    Type=simple
+    Restart=on-failure
+    ExecStart=/usr/local/bin/teleport start -c /etc/teleport.yaml
+
+    [Install]
+    WantedBy=multi-user.target
+  path: /etc/systemd/system/teleport.service
+- content: |
+    [Unit]
+    Description=Let's Encrypt renewal
+
+    [Service]
+    Type=oneshot
+    ExecStart=/usr/bin/certbot certonly --keep-until-expiring -n --agree-tos --email ${letsencrypt_email} --dns-route53 -d ${teleport_domain_name} --deploy-hook /usr/local/bin/teleport_enable_tls.sh
+  path: /etc/systemd/system/certbot.service
+- content: |
+    [Unit]
+    Description=Twice daily renewal of Let's Encrypt's certificates
+
+    [Timer]
+    OnCalendar=0/12:00:00
+    RandomizedDelaySec=1h
+    Persistent=true
+
+    [Install]
+    WantedBy=timers.target
+  path: /etc/systemd/system/certbot.timer
+- content: |
+    #!/bin/sh
+    # Enable TLS certificate in teleport if not already enabled
+    sed -i '/https_key_file:/s/^\(\s*\)#/\1/g' /etc/teleport.yaml
```

Comment:
This bit is not 100% clear to me. You just created the file above and you create a script to enable tls afterward? Is it required?

Reply:
I made this so if the first attempt to request a LE certificate fails, Teleport will start anyway with a self-signed certificate. Then once `certbot` is able to get a valid certificate, it'll enable it in Teleport and restart the service.

Reply:
I'll add that in the readme

Reply:
done

Reply:
Awesome thanks